src/librustc_metadata/cstore_impl.rs

```diff
@@ -128,6 +128,23 @@ provide! { <'tcx> tcx, def_id, cdata
         !cdata.is_proc_macro(def_id.index) &&
         cdata.maybe_entry(def_id.index).and_then(|item| item.decode(cdata).mir).is_some()
     }
+    is_const_fn => {
+        let constness = match cdata.entry(def_id.index).kind {
```

Comment:
@cramertj we had a discussion with @nikomatsakis about leaving any implementations that are longer than single liners as part of `CrateMetadata` and just calling those methods on `cdata` here.